src/Mapster/Utils/ReflectionUtils.cs

```diff
@@ -37,6 +38,24 @@ public static IEnumerable<IMemberModel> GetPublicFieldsAndProperties(this Type t
             return properties.Concat(fields);
         }
 
+        public static IEnumerable<IMemberModel> GetPrivateFieldsAndProperties(this Type type, bool allowNoSetter = true)
+        {
+            var properties = type.GetProperties(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Where(x => allowNoSetter || x.CanWrite)
+                .Select(CreateModel);
+
+            var fields = type.GetFields(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Where(x => (allowNoSetter || !x.IsInitOnly) && !IsBackingField(x))
+                .Select(CreateModel);
+
+            return properties.Concat(fields);
+        }
+
+        private static bool IsBackingField(FieldInfo field)
+        {
+            return field.CustomAttributes.Any(a => a.AttributeType == typeof(CompilerGeneratedAttribute));
+        }
```

Comment:
I already merged the 2 mentioned methods, but I'm not entirely sure what's the best way to access global settings from ReflectionUtils, should I just access it directly? It might create unwanted dependency there.